src/threadpoolwork-inl.h

```diff
@@ -24,23 +24,40 @@
 
 #if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
 
-#include "util-inl.h"
 #include "node_internals.h"
+#include "tracing/trace_event.h"
+#include "util-inl.h"
 
 namespace node {
 
 void ThreadPoolWork::ScheduleWork() {
   env_->IncreaseWaitingRequestCounter();
+  TRACE_EVENT_NESTABLE_ASYNC_BEGIN0(
+      TRACING_CATEGORY_NODE1(threadpoolwork), type_, this);
+  uv_work_cb fn = [](uv_work_t* req) {
+    ThreadPoolWork* self = ContainerOf(&ThreadPoolWork::work_req_, req);
+    self->DoThreadPoolWork();
+  };
+  if (*TRACE_EVENT_API_GET_CATEGORY_GROUP_ENABLED(
+          TRACING_CATEGORY_NODE1(threadpoolwork))) {
+    fn = [](uv_work_t* req) {
+      ThreadPoolWork* self = ContainerOf(&ThreadPoolWork::work_req_, req);
+      uint64_t start = uv_hrtime();
+      self->DoThreadPoolWork();
+      self->time_ = uv_hrtime() - start;
```

Comment:
Overall this LGTM but I'm curious if grabbing the time is really necessary. The trace event itself will have timing details relative to the overall trace.